docs/internals/contributing/writing-code/coding-style.txt

```diff
@@ -89,6 +89,26 @@ Python style
         """
         ...
 
+* In many places in Django's codebase, we catch low-level exceptions that have
+  been raised, and raise our own exception instead-- Also known as "wrapping
+  exceptions." The new exception will often have a different type and message.
+  That makes it easier to catch, understand and fix. Example::
+
+    try:
+        iter(patterns)
+    except TypeError as e:
+        msg = (
+            "The included URLconf '{name}' does not appear to have any "
+            "patterns in it. If you see valid patterns in the file then "
+            "the issue is probably caused by a circular import."
+        )
+        raise ImproperlyConfigured(msg.format(name=self.urlconf_name)) from e
+
+  In any place where an exception is wrapped, it's important to include the
+  ``from e`` part seen above. It causes Python to show the text
```

Comment:
`it's important to include the ``from e`` part as shown above. This coerces Python to display the message: "The above exception was the direct cause of the following exception:" in the traceback when chaining the above two exceptions.`

Reply:
`...when chaining multiple exceptions.` indicates to the reader that this concept can apply to more general situations beyond the example.

Reply:
Sounds like bikeshedding to me, sorry. The fact that "coerces" is totally weird aside, I don't really see "display the message" as better than "show the text", nor is it useful to introduce the user to the idea of more than 2 exceptions being chained in this esoteric point in the style guide.